fed/mosaic-sumo/src/test/java/org/eclipse/mosaic/fed/sumo/junit/SumoTraciRule.java

```diff
@@ -152,12 +157,24 @@ private void before() throws Throwable {
 
         redirectOutputToLog(); // this is necessary, otherwise TraCI will hang due to full output buffer
 
-        Thread.sleep(500); // wait a bit until TraCI server of SUMO is ready
-
         log.info("Connect to SUMO on port {}", port);
-        final Socket socket = new Socket("localhost", port);
-        socket.setPerformancePreferences(0, 100, 10);
-        socket.setTcpNoDelay(true);
+
+        Socket socket = null;
+        int tries = 0;
+        while (tries++ < MAX_CONNECTION_TRIES && socket == null) {
+            try {
+                Thread.sleep(500);
```

Comment:
Could also be a constant (e.g., `CONNECTION_RETRY_TIME`) and then be used in the catch-clause

Reply:
done